File: web/src/api/user-contact.ts
import request from './request'
import type { ApiResponse } from '@/types'

export interface UserContact {
  id: number
  user_id: number
  type: string // email, phone, feishu, wecom, dingtalk, webhook
  value: string
  name: string
  is_default: boolean
}

export const userContactApi = {
  list: () =>
    request.get<ApiResponse<UserContact[]>>('/user/contacts'),

  create: (data: { type: string; value: string; name: string; is_default?: boolean }) =>
    request.post<ApiResponse<UserContact>>('/user/contacts', data),

  update: (id: number, data: { type?: string; value?: string; name?: string; is_default?: boolean }) =>
    request.put<ApiResponse<UserContact>>(`/user/contacts/${id}`, data),

  delete: (id: number) =>
    request.delete<ApiResponse<null>>(`/user/contacts/${id}`),

  setDefault: (id: number) =>
    request.post<ApiResponse<null>>(`/user/contacts/${id}/default`),
}

File: internal/model/user_contact.go
package model

// UserContact stores a user's notification contact (email, phone, feishu, etc.).
type UserContact struct {
	BaseModel
	UserID    uint   `gorm:"index" json:"user_id"`
	Type      string `gorm:"size:32" json:"type"`     // email, phone, feishu, wecom, dingtalk, webhook
	Value     string `gorm:"size:256" json:"value"`   // the contact value
	Name      string `gorm:"size:64" json:"name"`     // user-friendly name
	IsDefault bool   `gorm:"default:false" json:"is_default"`
}

func (UserContact) TableName() string {
	return "user_contacts"
}
